### PR TITLE
Default for NetworkType is NATS

### LINKED ISSUE #3599: Default to NATS transport
Rather than default to libp2p, we should default to NATS as the network transport. This does not involve removing libp2p, only getting more exposure for the NATS transport. 

Ensure this is documented.

Ensure this is included in notes for next release.

### PR DESCRIPTION
**Pull request by rossjones**
Defaults to using NATS for the network transport if nothing is specified. This includes tests. There were quite a few required changes to cope with edge cases but should be good now.

We may want to merge this after the 1.3 release.

Fixes #3599

### CODE REVIEW

File: cmd/cli/node/action_test.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 type NodeActionSuite struct {
-	cmdtesting.BaseNATSSuite
+	cmdtesting.BaseSuite
```
Comment (wdbaruni) [inline cmd/cli/node/action_test.go:16]: I didn't know `BaseNATSSuite` existed!

**Review by wdbaruni — APPROVED** [review]

File: cmd/cli/serve/serve_test.go [hunk 2]
```diff
@@ -113,8 +113,10 @@ func (s *ServeSuite) serve(extraArgs ...string) (uint16, error) {
 
 	ctx, cancel := context.WithTimeout(s.ctx, maxServeTime)
 	errs, ctx := errgroup.WithContext(ctx)
+	s.T().Cleanup(func() {
+		cancel()
+	})
 
-	s.T().Cleanup(cancel)
```
[the remark this answers is not in the document]
Comment (rossjones) [reply cmd/cli/serve/serve_test.go:117]: None, other than the former leaves room for more debug info.  Can revert.

File: ops/aws/canary/lambda/pkg/test/local_test.go [hunk 3]
```diff
@@ -34,7 +34,8 @@ func TestScenariosAgainstDevstack(t *testing.T) {
 		nodeOverrides[i] = nodeOverride
 	}
 	stack := teststack.Setup(context.TODO(), t,
-		devstack.WithNumberOfHybridNodes(nodeCount),
+		devstack.WithNumberOfHybridNodes(1),
+		devstack.WithNumberOfComputeOnlyNodes(2),
```
[the remark this answers is not in the document]
Comment (rossjones) [reply ops/aws/canary/lambda/pkg/test/local_test.go:38]: This occurs in a few places, and it's a problem with setting up multiple hybrid nodes on the same machine, where the requesters all have their own NATS (not a shared one) and we get issues with both port usage and node registration where it looks like the compute nodes are not registering with all requester nodes.

File: cmd/cli/serve/serve_test.go [hunk 4]
```diff
@@ -173,7 +182,7 @@ func (s *ServeSuite) TestHealthcheck() {
 }
 
 func (s *ServeSuite) TestAPIPrintedForComputeNode() {
-	port, _ := s.serve("--node-type", "compute", "--log-mode", string(logger.LogModeStation))
+	port, _ := s.serve("--node-type", "compute,requester", "--log-mode", string(logger.LogModeStation))
```
[the remark this answers is not in the document]
Comment (rossjones) [reply cmd/cli/serve/serve_test.go:185]: Because the compute node needs to register somewhere
